koku/api/report/queries.py

```diff
@@ -240,16 +253,38 @@ def _check_for_operator_specific_filters(self, filter_collection):
             `.exclude`. Django adds "IS NOT NULL" when using `.exclude` which removes
             the `no-{option}` results.
         """
-        filter_collection = self._set_tag_filters(filter_collection)
-        filter_collection = self._set_operator_specified_tag_filters(filter_collection, "and")
-        filter_collection = self._set_operator_specified_tag_filters(filter_collection, "or")
-        tag_exclusion_composed = self._set_tag_exclusion_filters()
+        filter_collection = self._build_prefix_filters(filter_collection, TAG_PREFIX, self._mapper.tag_column)
+        filter_collection = self._set_operator_specified_prefix_filters(
+            filter_collection, TAG_PREFIX, self._mapper.tag_column, "and"
+        )
+        filter_collection = self._set_operator_specified_prefix_filters(
+            filter_collection, TAG_PREFIX, self._mapper.tag_column, "or"
+        )
+        tag_exclusion_composed = self._build_prefixed_exclusions(TAG_PREFIX, self._mapper.tag_column)
+        if self.is_aws and not self.is_openshift:
+            category_column = "cost_category"
+        elif self.is_aws and self.is_openshift:
+            category_column = "aws_cost_category"
+        else:
+            category_column = None
+            aws_category_exclusion_composed = None
+        if category_column:
+            filter_collection = self._build_prefix_filters(filter_collection, AWS_CATEGORY_PREFIX, category_column)
+            filter_collection = self._set_operator_specified_prefix_filters(
+                filter_collection, AWS_CATEGORY_PREFIX, category_column, "and"
+            )
+            filter_collection = self._set_operator_specified_prefix_filters(
+                filter_collection, AWS_CATEGORY_PREFIX, category_column, "or"
+            )
+            aws_category_exclusion_composed = self._build_prefixed_exclusions(AWS_CATEGORY_PREFIX, category_column)
```

Comment:
I'm not following this logic here. What does `AWS_CATEGORY_PREFIX` have to do with `category_column = "cost_category"`?